File: Backend/intoTheAbyssTestUtil/src/app/tiles/Stair.java
package app.tiles;

import app.utils.TileTypes;

public class Stair extends Tile {

    public Stair(){
        canHold = false;
        isPassable = true;
        type = TileTypes.STAIR;
    }

    @Override
    public TileTypes getType() {
        return type;
    }


}

File: Frontend/IntotheAbyss/app/src/main/java/com/example/intotheabyss/dungeonassets/Stair.kt
package com.example.intotheabyss.dungeonassets

import com.example.intotheabyss.utils.TileTypes

/**
 * This is a specific class Stair that inherits from the Tile superclass. This holds unique properties from Tile
 * and will be used to display stairs to the player,
 *
 * @constructor Creates a default stair type which is of TileType STAIR, cannot hold objects, and is passable
 * @author Benjamin Vogel
 */
class Stair: Tile() {
    override val type: TileTypes = TileTypes.STAIR

    init {
        canHold = false
        isPassable = true
    }
}